### PR TITLE
The Haskell Tool Stack

### PR DESCRIPTION
**Pull request by bjpbakker**
Build stack from source. See also #41787.

### CODE REVIEW

File: Library/Formula/stack.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  desc "The Haskell Tool Stack"
+  homepage "https://www.stackage.org"
+  url "https://github.com/commercialhaskell/stack/archive/v0.1.2.0.tar.gz"
+  sha256 "20ff0a36f773c2993e00c6f1bffaa33e881906d20f66cde0d557133842fc464c"
+
+  depends_on "ghc" => :build
+  depends_on "cabal-install" => :build
+
+  def install
+    system "cabal", "sandbox", "init"
```
Comment (mistydemeo) [inline Library/Formula/stack.rb]: We have some cabal helpers - take a look at pandoc for an example.

Comment (bjpbakker) [inline Library/Formula/stack.rb]: Awesome, thanks. I just updated the formula.

**Comment by mistydemeo** [comment]
Can you add the `setup_ghc_compilers` line below the dependencies too? Thanks!

**Comment by bjpbakker** [comment]
Thanks, I missed that.

**Comment by mistydemeo** [comment]
Test failed on one of the bots; reads like a parallelization issue when installing a copy of GHC during `stack setup`:

```
/usr/bin/install -c -m 755 -d "/private/tmp/stack20150731-57367-1jn5376/.stack/programs/x86_64-osx/ghc-7.8.4/bin"
/usr/bin/install -c -m 755 -d "/private/tmp/stack20150731-57367-1jn5376/.stack/programs/x86_64-osx/ghc-7.8.4/lib/ghc-7.8.4/include"
/usr/bin/install -c -m 755 -d "/private/tmp/stack20150731-57367-1jn5376/.stack/programs/x86_64-osx/ghc-7.8.4/lib/ghc-7.8.4/include"
install: mkdir /private/tmp/stack20150731-57367-1jn5376/.stack/programs/x86_64-osx/ghc-7.8.4: File exists
```

There are known issues with parallelized `make install` jobs of GHC; we wrap it in `ENV.deparallelize` in the GHC formula for this reason. Looks like it applies to `stack` if `MAKEFLAGS` is in the environment when it's run. You could wrap that line in an `ENV.deparallelize` block to work around it:

```ruby


  test do
    system "#{bin}/stack", "new"
    ENV.deparallelize { system "#{bin}/stack", "setup" }
    system "#{bin}/stack", "build"
  end
```

**Comment by mistydemeo** [comment]
Looks like the test timed out. Since it's downloading GHC, the amount of time this takes is variable, and we limit test runs to 5 minutes. Maybe leave the test at `system "#{bin}/stack", "new"`?

**Comment by mistydemeo** [comment]
Merged in 3d1c295f68e9c37440cd493062ba05c41bc75e28 with a simplified test and as `haskell-stack`. Thanks!

**Comment by bjpbakker** [comment]
Thanks for your help Misty.

**Pull request closed without merging**